tests/returntypes/transforms/test_mitigate_new.py

```diff
@@ -0,0 +1,570 @@
+
+    def test_integration(self):
+        """Test if the error of the mitigated result is less than the error of the unmitigated
+        result for a circuit with known expectation values"""
+        from mitiq.zne.scaling import fold_global
+        from mitiq.zne.inference import RichardsonFactory
+
+        noise_strength = 0.05
+
+        dev_noise_free = qml.device("default.mixed", wires=2)
+        dev = qml.transforms.insert(qml.AmplitudeDamping, noise_strength)(dev_noise_free)
+
+        n_wires = 2
+        n_layers = 2
+
+        shapes = qml.SimplifiedTwoDesign.shape(n_wires, n_layers)
+        np.random.seed(0)
+        w1, w2 = [np.random.random(s) for s in shapes]
+
+        def circuit(w1, w2):
+            qml.SimplifiedTwoDesign(w1, w2, wires=range(2))
+            qml.adjoint(qml.SimplifiedTwoDesign)(w1, w2, wires=range(2))
+            return qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliZ(1))
+
+        exact_qnode = qml.QNode(circuit, dev_noise_free)
+        noisy_qnode = qml.QNode(circuit, dev)
+
+        @qml.transforms.mitigate_with_zne([1, 2, 3], fold_global, RichardsonFactory.extrapolate)
+        @qml.qnode(dev)
+        def mitigated_qnode(w1, w2):
+            qml.SimplifiedTwoDesign(w1, w2, wires=range(2))
+            qml.adjoint(qml.SimplifiedTwoDesign)(w1, w2, wires=range(2))
+            return qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliZ(1))
+
+        exact_val = qml.math.stack(exact_qnode(w1, w2))
+        noisy_val = qml.math.stack(noisy_qnode(w1, w2))
+        mitigated_val = mitigated_qnode(w1, w2)
+
+        mitigated_err = np.abs(exact_val - mitigated_val)
+        noisy_err = np.abs(exact_val - noisy_val)
+
+        assert np.allclose(exact_val, [1, 1])
+        assert all(mitigated_err < noisy_err)
+
+    @pytest.mark.xfail(
+        reason="Using external tape transforms breaks differentiability",
+    )
+    def test_grad(self):
+        """Tests if the gradient is calculated successfully."""
+        from mitiq.zne.scaling import fold_global
+        from mitiq.zne.inference import RichardsonFactory
+
+        noise_strength = 0.05
+
+        dev_noise_free = qml.device("default.mixed", wires=2)
+        dev = qml.transforms.insert(qml.AmplitudeDamping, noise_strength)(dev_noise_free)
+
+        n_wires = 2
+        n_layers = 2
+
+        shapes = qml.SimplifiedTwoDesign.shape(n_wires, n_layers)
+        np.random.seed(0)
+        w1, w2 = [np.random.random(s, requires_grad=True) for s in shapes]
+
+        @qml.transforms.mitigate_with_zne([1, 2, 3], fold_global, RichardsonFactory.extrapolate)
+        @qml.qnode(dev)
+        def mitigated_circuit(w1, w2):
+            qml.SimplifiedTwoDesign(w1, w2, wires=range(2))
+            return qml.expval(qml.PauliZ(0))
+
+        g = qml.grad(mitigated_circuit)(w1, w2)
+        for g_ in g:
+            assert not np.allclose(g_, 0)
+
+
+# qnodes for the diffable ZNE error mitigation
+def qfunc(theta):
```

Comment:
Can we add some tests with two expecation values for the jacobian?

Reply:
Added jacobian tests for multi measurements.